packages/completer/src/handler.ts

```diff
@@ -349,34 +359,76 @@ export class CompletionHandler implements IDisposable {
     const state = this.getState(editor, position);
     const request: CompletionHandler.IRequest = { text, offset };
 
-    return this._connector
+    if ('responseType' in this._connector) {
+      return (this._connector as IDataConnector<
+        CompletionHandler.ICompletionItemsReply,
+        void,
+        CompletionHandler.IRequest
+      >)
+        .fetch(request)
```

Comment:
Would this work as a [user defined type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards)? Hopefully that would let us avoid the `as` b/c it would just infer the more specific type on this branch.